x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizerTests.java

```diff
@@ -4681,6 +4684,104 @@ public void testIsNullDisjunction() throws Exception {
         assertEquals(and, new PropagateNullable().rule(and));
     }
 
+    //
+    // Lookup
+    //
+
+    /**
+     * Expects
+     * <pre>{@code
+     * Limit[1000[INTEGER]]
+     * \_Join[JoinConfig[type=LEFT OUTER, unionFields=[int{r}#4]]]
+     *   |_EsqlProject[[...]]
+     *   | \_EsRelation[test][_meta_field{f}#13, emp_no{f}#7, first_name{f}#8, ge..]
+     *   \_LocalRelation[[int{f}#17, name{f}#18],[IntVectorBlock[vector=IntArrayVector[positions=10, values=[0, 1, 2, 3, 4, 5, 6, 7, 8,
+     * 9]]], BytesRefVectorBlock[vector=BytesRefArrayVector[positions=10]]]]
+     * }</pre>
+     */
+    public void testLookupSimple() {
+        var plan = optimizedPlan("""
+              FROM test
+            | RENAME languages AS int
+            | LOOKUP int_number_names ON int
+            """);
+        var join = as(plan, Join.class);
+
+        // Right is the lookup table
+        var right = as(join.right(), LocalRelation.class);
+        assertMap(
+            right.output().stream().map(Object::toString).sorted().toList(),
+            matchesList().item(containsString("int{f}")).item(containsString("name{f}"))
+        );
+
+        // Left is the rest of the query
+        var left = as(join.left(), EsqlProject.class);
+        assertThat(left.output().toString(), containsString("int{r}"));
+        var limit = as(left.child(), Limit.class);
+        assertThat(limit.limit().fold(), equalTo(1000));
+
+        // Join's output looks sensible too
+        assertMap(
+            join.output().stream().map(Object::toString).filter(s -> s.startsWith("int") || s.startsWith("name")).toList(),
+            matchesList()
+                /*
+                 * It's important that name is returned as a *reference* here
+                 * instead of a field. If it were a field we'd use SearchStats
+                 * on it and discover that it doesn't exist in the index. It doesn't!
+                 * We don't expect it to. It exists only in the lookup table.
+                 */
+                .item(containsString("name{r}"))
+                /*
+                 * Int is a reference here because we renamed it in project.
+                 * If we hadn't it'd be a field and that'd be fine.
+                 */
+                .item(containsString("int{r}"))
+        );
+    }
+
+    /**
+     * Expects
+     * <pre>{@code
+     * Limit[1000[INTEGER]]
+     * \_Aggregate[[name{f}#20],[MIN(emp_no{f}#10) AS MIN(emp_no), name{f}#20]]
+     *   \_Join[JoinConfig[type=LEFT OUTER, unionFields=[int{r}#4]]]
+     *     |_EsqlProject[[_meta_field{f}#16, ...]]
+     *     | \_EsRelation[test][_meta_field{f}#16, emp_no{f}#10, first_name{f}#11, ..]
+     *     \_LocalRelation[[name{f}#20, int{f}#21],[BytesRefVectorBlock[vector=BytesRefArrayVector[positions=10]], IntVectorBlock[vector=I
+     * ntArrayVector[positions=10, values=[0, 1, 2, 3, 4, 5, 6, 7, 8, 9]]]]]
+     * }</pre>
```

Comment:
We are not generating javadoc from these, so let's remove the `<pre>` tags to be consistent inside this file.
```suggestion
     * {@code
     * Limit[1000[INTEGER]]
     * \_Aggregate[[name{f}#20],[MIN(emp_no{f}#10) AS MIN(emp_no), name{f}#20]]
     *   \_Join[JoinConfig[type=LEFT OUTER, unionFields=[int{r}#4]]]
     *     |_EsqlProject[[_meta_field{f}#16, ...]]
     *     | \_EsRelation[test][_meta_field{f}#16, emp_no{f}#10, first_name{f}#11, ..]
     *     \_LocalRelation[[name{f}#20, int{f}#21],[BytesRefVectorBlock[vector=BytesRefArrayVector[positions=10]], IntVectorBlock[vector=I
     * ntArrayVector[positions=10, values=[0, 1, 2, 3, 4, 5, 6, 7, 8, 9]]]]]
     * }
```
Same for `PhysicalPlanOptimizerTests.java`